code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -954,6 +954,11 @@
 	gib()
 	return
 
+/mob/living/silicon/robot/update_canmove() // to fix lockdown issues w/ chairs
+    if(lockcharge)
+        canmove = 0

```

Comment:
I believe it is, in the return statement. The parent `update_canmove()` proc should handle that.